### PR TITLE
Add `dummy` factory methods for `ExecuteSuccessEvent` and `ExecuteFailureEvent`

### PR DESCRIPTION
**Pull request by hendrikmakait**
Addresses https://github.com/dask/distributed/pull/6676#discussion_r915815711

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/worker_state_machine.py [hunk 1]
```diff
@@ -70,6 +70,8 @@
 else:
     TaskStateState = str
 
+Type = type
```
Comment (gjoseph92) [inline distributed/worker_state_machine.py:73]: What mypy madness makes this necessary?

File: distributed/worker_state_machine.py [hunk 2]
```diff
@@ -746,6 +748,30 @@ def _after_from_dict(self) -> None:
...
 
+    @staticmethod
+    def dummy(
+        *,
+        key: str,
+        value: object = None,
+        start: float = 0.0,
+        stop: float = 1.0,
+        nbytes: int = 8,
+        type_: Type | None = None,
```
Comment (gjoseph92) [inline distributed/worker_state_machine.py:759]:
```suggestion
        type: Type | None = None,
```
I think ideally, `dummy` would have the same signature as the non-dummy `__init__`.

**Review by gjoseph92 — APPROVED** [review]

File: distributed/worker_state_machine.py (hunk 2 above)
Comment (hendrikmakait) [reply distributed/worker_state_machine.py]: Done, this was a remnant from `mypy` complaining about `type` being used as a type for `type`. I double-checked and switching the variable name is not a problem.

File: distributed/worker_state_machine.py (hunk 1 above)
Comment (hendrikmakait) [reply distributed/worker_state_machine.py:73]: Specifically, I think mypy is confused by `type` also being an instance variable on the dataclass.

Comment (crusaderky) [reply distributed/worker_state_machine.py:73]: but how comes it worked before?!?

File: distributed/worker_state_machine.py [hunk 3]
```diff
@@ -788,6 +814,32 @@ def from_exception(
...
+        stimulus_id: str,
+    ) -> ExecuteFailureEvent:
+        """Build a dummy event, with most attributes set to a reasonable default.
+        This is a convenience method to be used in unit testing only.
+        """
+        return ExecuteFailureEvent(
+            key=key,
+            start=start,
+            stop=stop,
+            exception=exception if exception is not None else Serialize(None),
```
Comment (crusaderky) [inline distributed/worker_state_machine.py:836]:
```suggestion
            exception=exception if exception is not None else Serialize(Exception()),
```

**Comment by crusaderky** [comment]
I removed the option to pass fields like exception, start, stop, etc. as they're all inconsequential to the worker state

**Review by crusaderky — APPROVED** [review]

File: distributed/worker_state_machine.py (hunk 1 above)
Comment (hendrikmakait) [reply distributed/worker_state_machine.py]: `mypy` only got confused in the scope of `dummy`, where it suddenly had 3 options for `type`:
* instance variable
* type class
* function parameter

**Pull request merged**